From 8a93c1d7234079a5bb33621ee88382f6d8587455 Mon Sep 17 00:00:00 2001
From: Simeon Marijon <simeon.marijon@foss.st.com>
Date: Thu, 21 Sep 2023 09:32:04 +0200
Subject: [PATCH 0516/1695] plat-stm32mp1: enable nvmem support for TAMP Backup
 registers

TAMP backup registers allow information sharing between different
software components/execution levels.
They are described individually as nvmem cells.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git a3cdcda8a4891935449e90574f5a5d6cf0eaccc9 plat-stm32mp1: enable nvmem support for TAMP Backup registers]
Change-Id: I8c91791daa44ccb15bbbeaa7ff6d2dc0f261b166
Signed-off-by: Simeon Marijon <simeon.marijon@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/328925
Reviewed-by: Etienne CARRIERE <etienne.carriere@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
(cherry picked from commit 02571dbc1b9eaa91df84739e44ccc2d680ac64d2)
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/351781
Tested-by: Simeon MARIJON <simeon.marijon@st.com>
Reviewed-by: Simeon MARIJON <simeon.marijon@st.com>
---
 core/arch/arm/plat-stm32mp1/conf.mk | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp1/conf.mk b/core/arch/arm/plat-stm32mp1/conf.mk
index 225b09071..459e5d07a 100644
--- a/core/arch/arm/plat-stm32mp1/conf.mk
+++ b/core/arch/arm/plat-stm32mp1/conf.mk
@@ -358,6 +358,8 @@ ifeq ($(CFG_TA_STM32MP_NVMEM),y)
 $(call force,CFG_STM32_BSEC_PTA,y,Required by CFG_TA_STM32MP_NVMEM)
 CFG_IN_TREE_EARLY_TAS += stm32mp_nvmem/1a8342cc-81a5-4512-99fe-9e2b3e37d626
 endif
+CFG_DRIVERS_NVMEM ?= y
+CFG_STM32_TAMP_NVRAM ?= y
 
 ifneq ($(CFG_WITH_SOFTWARE_PRNG),y)
 CFG_HWRNG_PTA ?= y
-- 
2.47.2

